### DISCUSSION TITLE
Uppercase SQL keywords only

### DISCUSSION THREAD

**Discussion by mittalyashu** [dbeaver/dbeaver#9426] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
One of the best way to read SQL queries is by formatting is properly and there's also need to be a proper differentiations between the SQL keywords and column names.

```sql
update vehicles_vehicle as v 
set price = s.price_per_vehicle
from shipments_shipment as s
where v.shipment_id = s.id 
```

> I know, there's also syntax highlighting.

**Describe the solution you'd like**
Add option to only uppercase the SQL keywords

**Describe alternatives you've considered**
DBeaver down-case them automatically on typing, so I have to copy-paste the query in backend and manually upper-case the SQL keywords.

**Additional context**

| Actual | Expected |
| --- | --- |
| <img width="236" alt="Screenshot 2020-08-01 at 8 50 57 PM" src="https://user-images.githubusercontent.com/29014463/89104686-c9f8bc00-d438-11ea-9b04-8c75f828899d.png"> | <img width="237" alt="Screenshot 2020-08-01 at 8 51 04 PM" src="https://user-images.githubusercontent.com/29014463/89104690-d1b86080-d438-11ea-95f5-ff0937654a9d.png"> |

**Comment by uslss** [comment, marked as answer]
Hi @mittalyashu ,
you could change this behaviour in preferences
![image](https://user-images.githubusercontent.com/31996417/89172128-d12fdf00-d58a-11ea-95b5-3226c7a8b6b4.png)

**Comment by mittalyashu** [comment]
Thanks for the update.

But, I think this `Keyword case` settings is bit confusing with
 
<img width="529" alt="Screenshot 2020-08-04 at 10 11 55 AM" src="https://user-images.githubusercontent.com/29014463/89253695-ef551800-d63a-11ea-9488-d2610796d1d9.png">

Even thought both of the have different functionality.

Also, can you format option "Uppercase keyword" in above dropdown.

**Comment by uslss** [comment]
To Upper/Lower Case options in Context menu are applied to selected/chosen text in SQL editor.
SQL Formatting preferences are applied globally within application/connection.
